### PR TITLE
add Metricly sink

### PR DESCRIPTION
**Pull request by yahonggu**
**What this PR does**

### CODE REVIEW

File: common/metricly/metricly.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+type MetriclyConfig struct {
+	ApiURL           string
+	ApiKey           string
+	ElementBatchSize int
+}
+
+func Config(uri *url.URL) (MetriclyConfig, error) {
+	config := MetriclyConfig{
+		ApiURL: "https://api.app.netuitive.com/ingest",
```
Comment (TheConnMan) [inline common/metricly/metricly.go]: `https://api.app.metricly.com/ingest`

File: metrics/sinks/metricly/metricly.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	var sent int
+	for i := 0; i < len(chunks); i++ {
+		sent += <-count
+	}
+	glog.Infof("Exported %d out of %d elements using %d workers", sent, len(elements), len(chunks))
+}
+
+func NewMetriclySink(uri *url.URL) (core.DataSink, error) {
+	config, _ := metricly.Config(uri)
+	glog.Info("Create metricly sink using config: ", config)
```
Comment (TheConnMan) [inline metrics/sinks/metricly/metricly.go]: Metricly should be capitalized.

File: metrics/sinks/metricly/metricly_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	//then
+	if len(elements) != 1 {
+		t.Errorf("there should be 1 element in elements, but actual =  %d", len(elements))
+	}
+	e := elements[0]
+	if e.Id != "namespace:kube-system/pod:heapster-ww9r6" {
+		t.Errorf("element id is wrong")
+	}
+	if len(e.Tags()) != 9 {
+		t.Errorf("element has wrong numbe of tags")
```
Comment (TheConnMan) [inline metrics/sinks/metricly/metricly_test.go]: `number` is misspelled.

Comment (TheConnMan) [reply metrics/sinks/metricly/metricly_test.go]: This is also missing a "the".

> Element has the wrong number of tags

or better yet

> The element has the wrong number of tags

File: metrics/sinks/metricly/metricly_test.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	}
+	if len(e.Metrics()) != 2 {
+		t.Errorf("element has wrong number of metrics")
```
Comment (TheConnMan) [inline metrics/sinks/metricly/metricly_test.go]: Same with these

**Review by TheConnMan — CHANGES_REQUESTED** [review]

File: common/metricly/metricly.go (hunk 1 above)
Comment (yahonggu) [reply common/metricly/metricly.go]: nice catch, I've searched both go-client and this repo and make sure there're no longer netuitive mentioned anywhere in the code

File: metrics/sinks/metricly/metricly.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+func (sink *MetriclyMetricsSink) Stop() {
+}
+
+type chunk struct {
+	start, end int
+}
+
+func (sink *MetriclyMetricsSink) ExportData(batch *core.DataBatch) {
+	glog.Info("Start exporting data batch to metricly ...")
```
Comment (TheConnMan) [inline metrics/sinks/metricly/metricly.go]: Metricly should be capitalized.

File: metrics/sinks/metricly/metricly.go [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+func (sink *MetriclyMetricsSink) Stop() {
+}
+
+type chunk struct {
+	start, end int
+}
+
+func (sink *MetriclyMetricsSink) ExportData(batch *core.DataBatch) {
```
Comment (TheConnMan) [inline metrics/sinks/metricly/metricly.go:47]: How large are these data batches? Trying to get an idea of how large the payloads will be and how often they'll come into API.

Additionally, is this a sync call to the sink from something else? Will a slowdown in a Metricly HTTP call cause downstream issues for a K8 cluster?

Comment (yahonggu) [reply metrics/sinks/metricly/metricly.go:47]: we're batching on number of elements (20 by default) per payload and send in parallel and the batch size is configurable. On average each element has about <20 metrics.

the caller of ExportData is a sync call and won't call the next Export until the current call returned.

Comment (TheConnMan) [reply metrics/sinks/metricly/metricly.go:47]: So high latency in the Metricly API or a large number of serial batched API calls could impact someones additional monitoring of K8?

Comment (yahonggu) [reply metrics/sinks/metricly/metricly.go:47]: No, I mean the call on our sink is a synced/blocking call. I don't think the call to multiple sinks are sequential.

Comment (TheConnMan) [reply metrics/sinks/metricly/metricly.go:47]: Ok, that's what I was concerned about. Thanks.

**Review by TheConnMan — APPROVED** [review]

**Pull request merged**